tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -106,6 +133,101 @@ static void seq_end_cb(int err, void *cb_data)
 	}
 }
 
+static void gatt_scann_cb(const bt_addr_le_t *addr, int8_t rssi,
+			  uint8_t adv_type, struct net_buf_simple *buf)
+{
+	static int cnt;
+	static int64_t buff_timestamp;
+	int64_t timestamp = k_uptime_get();
+	static int64_t delta;
+	uint8_t byte;
+	uint16_t word;
+
+	ASSERT_TRUE(gatt_param, "Gatt checker didn't get parameters");
+
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_IND) {
+		return;
+	}
+
+	if (gatt_param->service == MESH_PROV_SERVICE) {
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x0201, word);
+		/* flags */
+		byte = net_buf_simple_pull_u8(buf);
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x0303, word);
+		/* mesh provisioning service */
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x2718, word);
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x1516, word);
+		/* mesh provisioning service */
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x2718, word);
+	} else {
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x0201, word);
+		/* flags */
+		byte = net_buf_simple_pull_u8(buf);
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x0303, word);
+		/* mesh proxy service */
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x2818, word);
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x0c16, word);
+		/* mesh proxy service */
+		word = net_buf_simple_pull_be16(buf);
+		ASSERT_EQUAL(0x2818, word);
+		/* network ID */
+		byte = net_buf_simple_pull_u8(buf);
+		ASSERT_EQUAL(0x00, byte);
+	}
+
+	cnt++;
+
+	if (cnt > 1) {
+		delta = k_uptime_delta(&buff_timestamp);
+
+		LOG_INF("rx: %s +%d ms", txt_msg, delta);
+
+		ASSERT_TRUE(delta >= gatt_param->interval &&
+			delta < (gatt_param->interval + gatt_param->tolerance));
+	} else {
+		LOG_INF("rx: %s +%d ms", txt_msg, delta);
+	}
+
+	buff_timestamp = timestamp;
+
+	if (cnt == gatt_param->transmits) {
+		cnt = 0;
+		delta = 0;
+		buff_timestamp = 0;
+		LOG_INF("rx completed. stop observer.");
+		k_sem_give(&observer_sem);
+	}
+}
+
+static void rx_gatt_beacons(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type       = BT_HCI_LE_SCAN_PASSIVE,
+		.options    = BT_LE_SCAN_OPT_NONE,
+		.interval   = BT_MESH_ADV_SCAN_UNIT(1000),
+		.window     = BT_MESH_ADV_SCAN_UNIT(1000)
+	};
+	int err;
+
+	err = bt_le_scan_start(&scan_param, gatt_scann_cb);
+	ASSERT_FALSE(err && err != -EALREADY, "starting scan failed (err %d)", err);
+
+	k_sem_reset(&observer_sem);
```

Comment:
It is not wrong to have the k_sem_reset, but I believe it is not really necessary.

Reply:
removed